### PR TITLE
feat: get users list

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #5 
Implemented endpoint to get users list. As we have 3 auth type (from file, env or github) response consists of three sections:
`file` lists all users from credentials file, `env` provides user name set in env and `github` section shows whether github auth is enabled.

```json
{ "file": 
  {
    "Alice": {"email":"alice@example.com"},
    "John": {"groups":["reader","dev-lead"]}
  },
  "env": {"Jane":{}},
  "github": {"enabled":false}
}
```

### CODE REVIEW

**Review by g4s8 — CHANGES_REQUESTED** [review]
@olenagerasimova I'm not sure about the format of `/users` response: I think we need to hide implementation details about authentication methods and just return user list without specifying the source of these users (maybe we need another API for authentication's mechanisms, e.g. to enable or disable some particular authentication method). So I'm suggesting to keep current user list but moving to root of document and removing other auth-options configuration:
```json
  {
    "Alice": {"email":"alice@example.com"},
    "John": {"groups":["reader","dev-lead"]}
  }
```

**Comment by olenagerasimova** [comment]
@g4s8 ok, what about user from env auth (if it's enabled)? Should I add it to the list?

**Comment by g4s8** [comment]
> @g4s8 ok, what about user from env auth (if it's enabled)? Should I add it to the list?

@olenagerasimova environment variables are also just authentication method, user can be authenticated but don't have any permissions, so we need to add this user to any group either case. So no need to list environment logins here

**Comment by olenagerasimova** [comment]
> @olenagerasimova environment variables are also just authentication method, user can be authenticated but don't have any permissions, so we need to add this user to any group either case. So no need to list environment logins here

That is not quite correct: user from env variables is the same as user from credentials file (`email` and `groups` in this file are not required) and can have various permissions to various repositories (as permissions to access repository are set in repository settings file).

**Comment by olenagerasimova** [comment]
@g4s8 corrected, take a look, please

File: src/test/java/com/artipie/front/api/UsersTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * @since 0.1
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+class UsersTest {
+
+    @Test
+    void writesUsers() throws JSONException {
+        JSONAssert.assertEquals(
+            new Users(
+                Yaml.createYamlMappingBuilder().add(
```
Comment (g4s8) [inline src/test/java/com/artipie/front/api/UsersTest.java:27]: You can reuse classes from https://github.com/artipie/front/blob/master/src/test/java/com/artipie/front/settings/YamlCredentialsTest.java to build credentials (or maybe extract it to some test package)

**Review by g4s8 — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, looks good, just one comment about unit-test

**Comment by olenagerasimova** [comment]
@g4s8 thanks, corrected

**Review by g4s8 — APPROVED** [review]
@olenagerasimova thanks

**Pull request merged**